scattering/van_hove.py

```diff
@@ -214,3 +214,113 @@ def compute_partial_van_hove(trj, chunk_length=10, selection1=None, selection2=N
         g_r_t += g_r_t_frame
 
     return r, g_r_t
+
+def atom_conc_from_list(atom_list, trj):
+	""" 
+    	Generates a dictionary of atom concentration with the key corresponding to the atoms in atom_list
+
+    	Parameters
+    	----------
+    	trj : mdtraj.Trajectory
+        	trajectory on which partial vhf were calculated form
+	atom_list : list, optional, default = None
+        	list of unique atoms. if left at default, atom_list generated from trj
+
+
+    	Return
+    	-------
+    	mf : dict
+        	dictionary containing concentration of atoms from atom_list
+    	"""
+	
+	topology = trj.topology
+	atoms = [i.element.symbol for i in topology.atoms]
+    	
+	#atom_list generation
+	if atom_list == None:
+		atom_list = sorted(set(atoms))
+
+	mf = {}
+	atom_num_list = []
+    	
+	for i in range(len(atom_list)):
+        	atom_num_list.append(atoms.count(f"{atom_list[i]}"))
+        	mf[f'{atom_list[i]}'] = atom_num_list[i]/trj.n_atoms
+
+	return mf
+
+
+def form_factor_from_list(atom_list):
+    """ 
+    Generates a dictionary of atom form factors with keys corresponding to the atoms in atom_list
+
+    Parameters
+    ----------
+    atom_list : list
+        list of unique atoms
+
+
+    Return
+    -------
+    ff : dictionary
+        dictionary containing form factors of atoms from atom_list
+    """
+    
+    ff = {}
+    for i in range(len(atom_list)):
+            ff[f'{atom_list[i]}'] =  get_form_factor(element_name = f"{atom_list[i]}", water = None)
+    return ff
+
+
+def vhf_from_pvhf(trj, partial_dict,  atom_list = None):
+    """ 
+    Compute the total van Hove function from partial van Hove functions
+
+
+    Parameters
+    ----------
+    trj : mdtrj.Trajectory
+        trajectory on which partial vhf were calculated form
+    partial_dict : dictionary
+        dictionary containing partial vhf as a np.array. Key must be pairs listed in alphabetical order
+    atom_list : list, optional, default = None
+        list of unique atoms. if left at default, atom_list generated from trj
+
+
+
+    Return
+    -------
+    total_grt : nump.ndarray
+        Total Van Hove Function generated from addition of partial Van Hove Functions
+    """
+    
+    #atom_list generation
+
+    if atom_list == None:
+        atom_list = []
+        for i in range(topology.n_atoms):
+            if topology.atom(i).name not in atom_list:
+                atom_list.append(topology.atom(i).name)
+            atom_list = sorted(set(atom_list))
+
+    norm_factor = 0
+    coeff_list = []
+    combination = list(combinations_with_replacement(atom_list,2))
+    ff = form_factor_from_list(atom_list)
+    mf = atom_conc_from_list(atom_list, trj)
+    for pair in combination:
+        coeff = (ff[pair[0]] * ff[pair[1]] * mf[pair[0]] * mf[pair[1]])
+        coeff_list.append(coeff)
+        norm_factor = norm_factor + coeff
+    for i in range(len(combination)):
```

Comment:
Is it possible to combine these two forloops?  You may not have to have a `coeff_list`.  Instead you could use `coeff` to compute `temp_normalized` and then add to numpy array.